app/server/lib/DocApi.ts

```diff
@@ -1199,22 +1210,48 @@ export class DocWorkerApi {
       } else {
         parameters = req.body;
       }
-      if (parameters.workspaceId) { throw new Error('workspaceId not supported'); }
+
+      const documentName = optStringParam(parameters.documentName);
+      const workspaceId = optIntegerParam(parameters.workspaceId);
```

Comment:
Can you update optIntegerParam, it doesn't handle `NaN` case.

Reply:
It is possible to add document to a workspace that was moved to trash.

Reply:
@georgegevoian @berhalak in your opinions does this need a security review? Any thinking needed about whether someone could put a doc in someone else's workspace, that kind of thing?